src/pharmpy/model/random_variables.py

```diff
@@ -137,15 +148,35 @@ def subs(self, d):
         IIV_CL ~ N(0, OMEGA_NEW)
 
         """
-        mean = self._mean.subs(d)
-        variance = self._variance.subs(d)
+        mean = subs(self._mean, d)
+        variance = subs(self._variance, d)
         name = self._names[0]
         if name in d or sympy.Symbol(name) in d:
             name = d.get(name, d.get(sympy.Symbol(name)))
             if isinstance(name, sympy.Symbol):
                 name = name.name
         return NormalDistribution((name,), self._level, mean, variance)
 
+    def __getitem__(self, index):
```

Comment:
Interesting idea. What will be returned by indexing?

Reply:
The marginal!